### ISSUE TITLE
Inconsistent handling of output name

### ISSUE THREAD

**Issue by WhyNotHugo** [swaywm/sway#7796]
## Sway version

- **Sway Version:**
  - `sway version 1.8.1`

## Debug Log

https://paste.sr.ht/~whynothugo/e2b2cfece5667ae58d415e380e2fe0875d17fe0c

## Description

I am configuring both a scale and wallpaper for my laptop's output.

```console
> swaymsg -t get_outputs
Output eDP-1 'IVO 0x057D Unknown' (focused)
  Current mode: 1920x1080 @ 60.010 Hz
  Position: 0,0
  Scale factor: 1.000000
  Scale filter: nearest
  Subpixel hinting: unknown
  Transform: normal
  Workspace: 1 1:一
  Max render time: off
  Adaptive sync: disabled
  Available modes:
    1920x1080 @ 60.010 Hz
    1920x1080 @ 47.999 Hz

> wlr-randr
eDP-1 "IVO 0x057D (eDP-1)"
  Make: IVO
  Model: 0x057D
  Serial: (null)
  Physical size: 310x170 mm
  Enabled: yes
  Modes:
    1920x1080 px, 60.009998 Hz (preferred, current)
    1920x1080 px, 47.999001 Hz
  Position: 0,0
  Transform: normal
  Scale: 1.000000
  Adaptive Sync: disabled
```

I have tried two different approaches:

The first approach applies only the wallpaper, but not the scale:
```
output "IVO 0x057D" {
  scale 1.5
  bg ~/pictures/wallpapers/destiny.current fill
}
```

The second approach applies only the scale and not the wallpaper:
```
output "IVO 0x057D Unknown" {
  scale 1.5
  bg ~/pictures/wallpapers/destiny.current fill
}
```

This approach works, but obviously this shouldn't be the correct appoach:

```
output "IVO 0x057D" {
  bg ~/pictures/wallpapers/destiny.current fill
}
output "IVO 0x057D Unknown" {
  scale 1.5
}
```

**Comment by emersion** [comment]
Right. This is an inconsistency in wlroots' output description and Sway's output identifier.

Ideally we'd pick a more robust way to refer to outputs in Sway, and somehow expose it via Wayland...

**Comment by WhyNotHugo** [comment]
A wl_output wayland object has:

- name
- description
- make
- model

wlroots' wlr_output struct has:

- name
- description
- make
- model
- serial (I _think_ that this is the device serial, but not 100% sure)

It seems that sway is the odd one out here, so intuitively, I think that the
one that should change is sway itself.

Perhaps something like this?:

```
output --name IVO --model 0x057D {
  scale 1.5
  bg ~/pictures/wallpapers/destiny.current fill
}
```

Or maybe?:

```
output name=IVO model=0x057D {
  scale 1.5
  bg ~/pictures/wallpapers/destiny.current fill
}
```

**Comment by WhyNotHugo** [comment]
Not 100% sure, but I have the impression that the above syntax (`--name XXX`) can be implemented in a way that is backwards compatible with the current implementation.

So maybe the new syntax can be introduced in 1.9 or 1.10, and the old one dropped a release or two later.

**Comment by emersion** [comment]
I'd prefer to generate a more persistent output identifier instead. See https://gitlab.freedesktop.org/emersion/libdisplay-info/-/merge_requests/115

**Comment by WhyNotHugo** [comment]
I guess that we'll wait for that to be merged before proceeding here?
